src/files-mfs/read-pull-stream.js

```diff
@@ -25,37 +23,24 @@ module.exports = (common, options) => {
 
     after(() => common.teardown())
 
-    it('should not read not found, expect error', (done) => {
+    it('should not read not found, expect error', () => {
       const testDir = `/test-${hat()}`
 
-      pull(
-        ipfs.files.readPullStream(`${testDir}/404`),
-        collect((err) => {
-          expect(err).to.exist()
-          expect(err.message).to.contain('does not exist')
-          done()
-        })
-      )
+      return expect(pullToPromise.any(ipfs.files.readPullStream(`${testDir}/404`))).to.eventually.be.rejected
+        .and.be.an.instanceOf(Error)
+        .and.have.property('message')
+        .that.include('does not exist')
     })
 
-    it('should read file', (done) => {
+    it('should read file', async () => {
       const testDir = `/test-${hat()}`
 
-      series([
-        (cb) => ipfs.files.mkdir(testDir, cb),
-        (cb) => ipfs.files.write(`${testDir}/a`, Buffer.from('Hello, world!'), { create: true }, cb)
-      ], (err) => {
-        expect(err).to.not.exist()
-
-        pull(
-          ipfs.files.readPullStream(`${testDir}/a`),
-          collect((err, bufs) => {
-            expect(err).to.not.exist()
-            expect(bufs).to.eql([Buffer.from('Hello, world!')])
-            done()
-          })
-        )
-      })
+      await ipfs.files.mkdir(testDir)
+      await ipfs.files.write(`${testDir}/a`, Buffer.from('Hello, world!'), { create: true })
+
+      const bufs = await pullToPromise.any(ipfs.files.readPullStream(`${testDir}/a`))
```

Comment:
IMO pull-streams pipelines should not be transformed in promises, you lose flexibility and gain 0 value. Pull-streams should either disappear in the future or be left to work as they were created.